pygfx/__init__.py

```diff
@@ -23,35 +23,61 @@
 
 # Temp fix for pyinstaller to pick up pylinalg
 import pylinalg
-
 del pylinalg
 
+
 __version__ = "0.5.0"
 version_info = tuple(map(int, __version__.split(".")))
 
-__wgpu_version_range__ = "0.17.3", "0.19.0"
-__pylinalg_version_range__ = "0.4.1", "0.5.0"
-
 
-def _check_lib_version(libname, pipname, version_range):
+def _get_dependency_version_ranges():
+    # Try import, dont care when this fails (e.g. frozen or py < 3.11)
+    try:
+        import os, tomllib, importlib.metadata  # noqa
+    except ImportError:
+        return {}
+    # Load versions
+    this_dir = os.path.dirname(os.path.abspath(__file__))
+    pyproject_file = os.path.join(this_dir, "..", "pyproject.toml")
+    if os.path.isfile(pyproject_file):
+        with open(pyproject_file, "rb") as fp:
+            dependencies = tomllib.load(fp)['project']['dependencies']
+    else:
+        dependencies = importlib.metadata.requires("pygfx")
```

Comment:
I don't fully understand why you need to first attempt to read the toml file? Isn't importlib.metadata.requires enough?

Reply:
Good question.

We have version ranges for wgpu-py and pylinalg, for which we know pygfx works. Pip respects them, so for end-users we don't really have to do this check. This check is therefore mosly for people who use bleeding edge versions of wgpu and/or pygfx. It's a nice warning in case you've updated one, but not the other.

The problem with `importlib.metadata.requires("pygfx")` is that it returns the values as they were the last time you did `pip install -e .`. So basically worthless for this scenario.

Having written this down, I think we can simply only perform the check when there's a pyproject file, bc that's a good indication pygfx is installed in editable/dev mode.